client/packages/react/src/useQuery.ts

```diff
@@ -33,17 +43,17 @@ export function useQuery<
   const query = _query ? coerceQuery(_query) : null;
   const queryHash = weakHash(query);
 
-  // (XXX): We use a ref to store the result of the query because `useSyncExternalStore`
-  // uses `Object.is` to compare the previous and next state.
+  // We use a ref to store the result of the query.
+  // This is becuase `useSyncExternalStore` uses `Object.is`
+  // to compare the previous and next state.
   // If we don't use a ref, the state will always be considered different, so
   // the component will always re-render.
-  const resultCacheRef =
-    useRef<LifecycleSubscriptionState<Q, Schema, WithCardinalityInference>>(
-      defaultState,
-    );
+  const resultCacheRef = useRef<
+    LifecycleSubscriptionState<Q, Schema, WithCardinalityInference>
+  >(stateForResult(_core._reactor.getPreviousResult(query)));
```

Comment:
Technically, this means that we _could_ skip the first 'callback' for subscribeQuery. Reason: since we already cached the previous result here, `subscribeQuery` would in-effect return the same cached result in the first callback response.

But, it's not 100% true. Something _could_ have happened between the useEffect call and this cached call. It would be amazing if we could refactor Reactor, so we could have more stable values for query results. This way, we would know exactly _when_ it was okay to skip an update. i.e if `store` hasn't changed, and `transactions queue` has not changed.